app/src/main/java/de/xikolo/controllers/channels/ChannelDetailsFragment.java

```diff
@@ -134,23 +135,29 @@ public boolean onOptionsItemSelected(MenuItem item) {
     public void setupView(Channel channel) {
         if (getActivity() instanceof ChannelDetailsActivity) {
             layoutHeader.setVisibility(View.GONE);
-        }
-        else {
+        } else {
             GlideApp.with(this).load(channel.imageUrl).into(imageChannel);
             textTitle.setText(channel.title);
         }
 
-        if(courseListAdapter != null)
+        if (courseListAdapter != null)
             courseListAdapter.setButtonColor(channel.getColorOrDefault());
     }
 
     @Override
     public void showCourseList(SectionList<String, List<Course>> courses) {
-        if(courseListAdapter != null)
+        if (courseListAdapter != null)
             courseListAdapter.update(courses);
 
-        if(scrollToCourses)
-            courseList.scrollToPosition(1);
+        if (scrollToCoursePosition >= 0) {
+            try {
+                ChannelDetailsActivity a = ((ChannelDetailsActivity) getActivity());
+                if (a != null)
+                    a.appBarLayout.setExpanded(false);
+                courseList.smoothScrollToPosition(scrollToCoursePosition + 3);
```

Comment:
What does the magic number `3` here?

Reply:
description + heading + (n+1)th element (as the first element is 1 and not 0)

Reply:
Would be a good comment on top of this line :)